src/dc/nipals.js

```diff
@@ -0,0 +1,64 @@
+
+/**
+ * NIPALS
+ * @param {Matrix} X
+ * @param {*} Y
+ * @param {*} u
+ */
+export function nipals(X, options = {}) {
+  const { Y } = options;
+  let u;
+  if (Y) {
+    u = Y.getColumnVector(0);
+  } else {
+    u = X.getColumnVector(0);
+  }
+
+  let diff = 1;
+  let t, q, w, tOld;
+
+  let counter = 0;
+  for (var i = 0; diff > 1e-10; i++) {
```

Comment:
Should also be an options with default value

Reply:
code has been updated as suggested: options.terminationCriteria has been set with default to 1e-10